### ISSUE TITLE
[New Models] Reflect the change in dataset on the dataloader during training

### ISSUE THREAD

**Issue by YaohouF** [open-mmlab/mmocr#2057]
### Model/Dataset/Scheduler description

Hi there

I am currently trying to use copy-paste style data augmentation on the training dataset for pre-training. It copies objects from other dataset to the training images, so the annotations are changing constantly. 
This is a large and complex library, so wonder if I can get some suggestions or clues on which part of the code should I modify to make it works?

### Open source status

- [ ] The model implementation is available
- [ ] The model weights are available.

**Comment by gaotongxiao** [comment]
You can create a custom transform from `BaseTransfrom` and use your external library in its `transform` method. Here are some examples and docs:

https://github.com/open-mmlab/mmocr/blob/2caab0a4e7dff2929ee6113927098384af6072bf/mmocr/datasets/transforms/textdet_transforms.py#L215

https://mmocr.readthedocs.io/en/dev-1.x/basic_concepts/transforms.html

**Comment by YaohouF** [comment]
> You can create a custom transform from `BaseTransfrom` and use your external library in its `transform` method. Here are some examples and docs:
> 
> https://github.com/open-mmlab/mmocr/blob/2caab0a4e7dff2929ee6113927098384af6072bf/mmocr/datasets/transforms/textdet_transforms.py#L215
> 
> https://mmocr.readthedocs.io/en/dev-1.x/basic_concepts/transforms.html

Thank you so much for the comment:) Will quickly check it out.
